src/App.js

```diff
@@ -22,34 +22,58 @@ export function App () {
         FILE_VIEW: 1,
     };
 
-    const urlSearchParams = new URLSearchParams(window.location.search, "?");
-
     const [fileInfo, setFileInfo] = useState(null);
     const [theme, setTheme] = useState(THEME_STATES.DARK);
     const [appMode, setAppMode] = useState(null);
+    const [prettify, setPrettify] = useState(null);
+    const [logEventIdx, setLogEventIdx] = useState(null);
 
     const switchTheme = (theme) => {
         localStorage.setItem("ui-theme", theme);
         document.getElementById("app").setAttribute("data-theme", theme);
         setTheme(theme);
     };
 
-    useEffect(() => {
-        console.debug("Version:", config.version);
-        const lsTheme = localStorage.getItem("ui-theme");
-        switchTheme(lsTheme === THEME_STATES.LIGHT?THEME_STATES.LIGHT:THEME_STATES.DARK);
+    /**
+     * Loads the initial state of the viewer from url arguments.
+     *
+     * ### Example:
+     * localhost:3010/?filePath=/logs/sample.clp.zst&prettify=true&logEventIdx=1
+     *
+     * @return {[string,boolean,string]} [filePath,prettify,logEventIdx]
+     */
+    const getInitialStateFormUrl = () => {
+        const urlSearchParams = new URLSearchParams(window.location.search, "?");
+        const hash = window.location.hash;
+        const filePath = urlSearchParams.get("filePath");
+        const prettify = urlSearchParams.get("prettify") === "true";
+        const logEventIdx = (hash.includes("logEventIdx") ?hash.split("=").pop():null);
```

Comment:
* This will break if there's more than one hash parameter in the URL.
* Style should format this with spaces before and after each operator (`?`, `:`).

Reply:
Done